homeassistant/data_entry_flow.py

```diff
@@ -352,6 +352,18 @@ async def async_configure(
     ) -> _FlowResultT:
         """Continue a data entry flow."""
         result: _FlowResultT | None = None
+
+        # Workaround for flow handlers which have not been upgraded to pass a show
+        # progress task, needed because of the change to eager tasks in HA Core 2024.5,
+        # can be removed in HA Core 2024.8.
+        flow = self._progress.get(flow_id)
+        if flow and flow.deprecated_show_progress:
+            if (cur_step := flow.cur_step) and cur_step[
+                "type"
+            ] == FlowResultType.SHOW_PROGRESS:
+                # Allow the progress task to finish befere we call the flow handler
+                await asyncio.sleep(0.5)
```

Comment:
```suggestion
                await asyncio.sleep(0)
```

Will this work with sleep 0?

Reply:
I don't agree with that.
Before the change to eager tasks, a newly created task would not have a chance to start executing until the next yield.

The situation now is that the newly created task is already executing, I don't think a `sleep(0)` is guaranteed to allow the task who created us to execute before the newly created task is executed again, unless the asyncio task scheduling is strictly round-robin.